### PR TITLE
Follow up on MEX interface

### PR DESCRIPTION
**Pull request by FreyJo**
- [x] sqp_rti test
- [x] test constraints on algebraic variables in matlab/octave
- [x] print stat routine for ocp solver
- [x] fix `linear_mass_spring_model example_ocp`
- [x] octave test
  - [x] with nonuniform grid
  - [x] with qpOASES
  - [x] linear least squares cost in algebraic variables

### CODE REVIEW

File: interfaces/acados_matlab_octave/ocp_set_ext_fun.m [hunk 1]
```diff
@@ -421,6 +421,7 @@
 			mex(acados_include, acados_interfaces_include, external_include, blasfeo_include,...
 				hpipm_include, acados_lib_path, acados_matlab_octave_lib_path, model_lib_path, '-lacados',...
 				 '-lhpipm', '-lblasfeo', ['-l', model_name], mex_files{1});
+			setenv('CFLAGS', '');
```
Comment (giaf) [inline interfaces/acados_matlab_octave/ocp_set_ext_fun.m]: why this?

File: interfaces/acados_matlab_octave/sim_set_ext_fun.m [hunk 2]
```diff
@@ -179,6 +180,7 @@
 			cflags_tmp = [cflags_tmp, ' -DFUN_NAME=', fun_names{ii}];
 			setenv('CFLAGS', cflags_tmp);
 			mex(acados_include, acados_interfaces_include, acados_lib_path, acados_matlab_octave_lib_path, model_lib_path, '-lacados', '-lhpipm', '-lblasfeo', ['-l', model_name], fullfile(acados_mex_folder, 'sim_set_ext_fun_gen.c'));
+			setenv('CFLAGS', '');
```
Comment (giaf) [inline interfaces/acados_matlab_octave/sim_set_ext_fun.m]: also here, why this?

**Comment by giaf** [comment]
weird that there are also commits from my PR, they should be already merged into master

File: interfaces/acados_matlab_octave/ocp_set_ext_fun.m (hunk 1 above)
Comment (FreyJo) [reply interfaces/acados_matlab_octave/ocp_set_ext_fun.m]: you are right, they are not needed anymore, was connected to the qpOASES flag issue.

**Comment by FreyJo** [comment]
> weird that there are also commits from my PR, they should be already merged into master

yes. I merged your PR into my branch to be up to date. I guess/hope it is just a bug in how the PR is shown somehow..

**Comment by giaf** [comment]
> > weird that there are also commits from my PR, they should be already merged into master
> 
> yes. I merged your PR into my branch to be up to date. I guess/hope it is just a bug in how the PR is shown somehow..

maybe try to merge the master into the PR branch first

**Comment by giaf** [comment]
ok it looks good for me, merging now!

**Pull request merged**